src/main/java/bio/terra/service/dataset/flight/ingest/IngestCopyControlFileStep.java

```diff
@@ -49,4 +49,10 @@ public StepResult doOptionalStep(FlightContext context)
     workingMap.put(IngestMapKeys.INGEST_CONTROL_FILE_PATH, newPath);
     return StepResult.getStepResultSuccess();
   }
+
+  @Override
+  public StepResult undoOptionalStep(FlightContext context) {
+    IngestUtils.deleteScratchFile(context, gcsPdao);
```

Comment:
There's a couple of places where you call this in an undo.  I just want to make sure: is it possible that deleteScratchFIle might fail if it can't find a bucket (e.g. it's partially loaded?). In that case, this should catch the exception since this likely fail if the copy step failed partially through, no?

Reply:
I don't think so. The bucket is created in a completely different step, so if we hit this undo step, the bucket is definitely there. If the bucket can't be created, this step would never run.

I think we do want to do this for azure too. Was leaving that alone since there were a lot of code changes in azure-land.